src/interface_adapter/recipes_getter/GetRecipesPresenter.java

```diff
@@ -0,0 +1,39 @@
+package interface_adapter.recipes_getter;
+
+import interface_adapter.ViewManagerModel;
+import interface_adapter.add_item.AddItemViewModel;
+import interface_adapter.logged_in.LoggedInState;
+import interface_adapter.logged_in.LoggedInViewModel;
+import use_case.recipes_getter.GetRecipeOutputBoundary;
+import use_case.recipes_getter.GetRecipeOutputData;
+
+public class GetRecipesPresenter implements GetRecipeOutputBoundary {
+
+    private final GetRecipesViewModel getRecipesViewModel;
+
+    private final LoggedInViewModel loggedInViewModel;
+
+    private ViewManagerModel viewManagerModel;
+
+    public GetRecipesPresenter(GetRecipesViewModel getRecipesViewModel, LoggedInViewModel loggedInViewModel) {
+        this.getRecipesViewModel = getRecipesViewModel;
+        this.loggedInViewModel = loggedInViewModel;
+    }
+    @Override
+    public void prepareSuccessView(GetRecipeOutputData results) {
+        GetRecipesState getRecipesState = getRecipesViewModel.getState();
+        getRecipesState.setRecipes(results.ReturnRecipe());
+        this.getRecipesViewModel.setState(getRecipesState);
+        this.getRecipesViewModel.firePropertyChanged();
+
+        this.viewManagerModel.setActiveView(getRecipesViewModel.getViewName());
+        this.viewManagerModel.firePropertyChanged();
```

Comment:
consider creating a package called 'Recipe Result' or something like that, and add 'ResultState' and 'ResultViewModel' under that package. Success cases should switch to Result view. Consider using 'LoginPresenter' from week5 coding as a reference.